File: Java/Tests/jdi-uitest-tutorialtests/src/test/java/com/epam/jdi/uitests/testing/career/common/tests/simple/examples/TableExamples.java
package com.epam.jdi.uitests.testing.career.common.tests.simple.examples;

import com.epam.commons.map.MapArray;
import com.epam.jdi.uitests.testing.career.common.tests.TestsBase;
import com.epam.jdi.uitests.testing.career.page_objects.dataProviders.AttendeeProvider;
import com.epam.jdi.uitests.testing.career.page_objects.entities.Attendee;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.interfaces.ICell;
import com.epam.web.matcher.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.io.IOException;
import java.lang.reflect.Method;

import static com.epam.jdi.uitests.testing.career.page_objects.site.EpamSite.jobDescriptionPage;
import static com.epam.jdi.uitests.testing.career.page_objects.site.EpamSite.jobListingPage;
import static com.epam.jdi.uitests.web.selenium.elements.complex.table.Column.inColumn;
import static com.epam.jdi.uitests.web.selenium.elements.complex.table.WithValue.withValue;


public class TableExamples extends TestsBase {
    @BeforeMethod
    public void before(Method method) throws IOException {
        jobListingPage.isOpened();
        Assert.isFalse(jobListingPage.jobsList::isEmpty);
    }

    @Test
    public void searchInTableExample() {
        jobListingPage.jobsList.row(withValue("Senior QA Automation Engineer"), inColumn("JOB_NAME"))
            .get("APPLY").select();

        jobDescriptionPage.checkOpened();
    }

    @Test
    public void searchByMultiCriteriaInTableExample() {
        MapArray<String, ICell> firstRow = jobListingPage.jobsList.rows(
                "JOB_NAME=Senior QA Automation Engineer",
                "JOB_CATEGORY=Software Test Engineering")
                .first().value;
        firstRow.get("APPLY").select();

        jobDescriptionPage.checkOpened();
    }
}

File: C#.Net/Tests/JDI Tests/Tests/Epam/TableExamples.cs
using System.Linq;
using JDI_Web.Selenium.Elements.Complex.Table;
using NUnit.Framework;
using static JDI_Tests.Epam_UIObjects.EpamSite;

namespace JDI_Tests.Tests.Epam
{
    [TestFixture]
    public class TableExamples
    {
        [SetUp]
        public void SetUp()
        {
            JobListingPage.IsOpened();
            Assert.That(JobListingPage.JobsList.Empty, Is.False);
        }

        [Test]
        public void GetTableInfoExample()
        {
            Assert.That(JobListingPage.JobsList.Columns.Count, Is.EqualTo(4));
            Assert.That(JobListingPage.JobsList.Rows.Count, Is.EqualTo(2));
            Assert.That(JobListingPage.JobsList.Count, Is.EqualTo(2));
            var jobListValues = "||X||Title|Type|Location|Apply||\n" +
                                   "||1||QA Specialist|Software Test Engineering|St-Petersburg, Russia|Apply||\n" +
                                   "||2||Senior QA Automation Engineer|Software Test Engineering|St-Petersburg, Russia|Apply||";
            Assert.That(JobListingPage.JobsList.Value, Is.EqualTo(jobListValues));
        }

        [Test]
        public void SearchInTableExample()
        {
            JobListingPage.JobsList.Row("Senior QA Automation Engineer", Column.column("Title")).Apply.Click();
             JobDescriptionPage.CheckOpened();
        }

        [Test]
        public void SearchByMultiCriteriaInTableExample()
        {
            var firstRow = JobListingPage.JobsList.GetRows(
                "Title=Senior QA Automation Engineer",
                "Type=Software Test Engineering").First();
            Assert.That(firstRow.Location.GetText, Is.EqualTo("St-Petersburg, Russia"));
        }
    }
}
